### PR TITLE
Migrate GEM/CSC validation to esConsumes

### PR DESCRIPTION
**Pull request by dildick**
#### PR description:

Migrate GEM and CSC validation to esConsumes and edm::ESGetToken to provide better threading efficiency, in response to #31061.

Included: 
RecoLocalMuon/CSCEfficiency
RecoLocalMuon/CSCValidation
Validation/CSCRecHits
Validation/MuonCSCDigis
Validation/MuonGEMDigis
Validation/MuonGEMHits
Validation/MuonGEMRecHits
Validation/MuonHits
Validation/MuonME0Validation

#### PR validation:

Tested with WF 23234.0

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by Dr15Jones** [comment]
Thanks!

File: Validation/MuonCSCDigis/plugins/CSCDigiValidation.cc [hunk 1]
```diff
@@ -56,8 +55,7 @@ void CSCDigiValidation::analyze(const edm::Event &e, const edm::EventSetup &even
   theSimHitMap.fill(e);
 
   // find the geometry & conditions for this event
-  edm::ESHandle<CSCGeometry> hGeom;
-  eventSetup.get<MuonGeometryRecord>().get(hGeom);
+  edm::ESHandle<CSCGeometry> hGeom = eventSetup.getHandle(geomToken_);
```
Comment (Dr15Jones) [inline Validation/MuonCSCDigis/plugins/CSCDigiValidation.cc:58]: You could make your life a bit easier by doing
```C++
const CSCGeometry *pGeom = &eventSetup.getData(geomToken_);
```

**Comment by dildick** [comment]
@Dr15Jones I don't have it running yet. Trying to figure out why this keeps happening in WF 23234.0

```
----- Begin Fatal Exception 09-Aug-2020 00:27:05 CDT-----------------------
An exception of category 'ESGetTokenWrongTransition' occurred while
   [0] Processing  stream begin Run run: 1 stream: 0
   [1] Calling method for module GEMRecHitValidation/'gemRecHitsValidation'
Exception Message:
The transition ID stored in the ESGetToken does not match the
transition where the token is being used. The associated record
type is: MuonGeometryRecord
For producers, filters and analyzers this transition ID is
set as a template parameter to the call to the esConsumes
function that creates the token. Event is the default transition.
Other possibilities are BeginRun, EndRun, BeginLuminosityBlock,
or EndLuminosityBlock. You may need multiple tokens if you want to
get the same data in multiple transitions. The transition ID has a
different meaning in ESProducers. For ESProducers, the transition
ID identifies the function that produces the EventSetup data (often
there is one function named produce but there can be multiple
functions with different names). For ESProducers, the ESGetToken
must be used in the function associated with the ESConsumesCollector
returned by the setWhatProduced function.
----- End Fatal Exception -------------------------------------------------
```

File: Validation/MuonGEMRecHits/plugins/GEMRecHitValidation.cc [hunk 2]
```diff
@@ -12,12 +12,13 @@ GEMRecHitValidation::GEMRecHitValidation(const edm::ParameterSet& pset)
   const auto& simhit_pset = pset.getParameterSet("gemSimHit");
   const auto& simhit_tag = simhit_pset.getParameter<edm::InputTag>("inputTag");
   simhit_token_ = consumes<edm::PSimHitContainer>(simhit_tag);
+  geomToken_ = esConsumes<GEMGeometry, MuonGeometryRecord>();
```
Comment (Dr15Jones) [inline Validation/MuonGEMRecHits/plugins/GEMRecHitValidation.cc:15]: You will need two different tokens, one for getting the data during the BeginRun and the other for getting it during the Event. When doing the esConsumes call for the BeginRun token you will use a third template argument edm::Transition::BeginRun

**Comment by dildick** [comment]
Step 3 of WF 23234.0 is now running. Let me check if also step 4 is running.

**Comment by dildick** [comment]
@Dr15Jones 

```
23234.0_TTbar_14TeV+2026D49+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Mon Aug 10 10:54:28 2020-date Mon Aug 10 10:30:50 2020; exit: 0 0 0 0
1 1 1 1 tests passed, 0 0 0 0 failed
```

**Comment by dildick** [comment]
@ptcox FYI, this PR also applies the esConsumes to RecoLocalMuon/CSCEfficiency and RecoLocalMuon/CSCValidation. You'll notice I also cleaned up the header files - there were quite a number of unnecessary include statements.

**Comment by ptcox** [comment]
Thanks. Good work!

**Comment by perrotta** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by jpata** [comment]
+1

- tests pass with no changes to outputs, as expected for a technical change
- moves to `esConsumes` in a number of packages
- changes cout -> edm::LogInfo in a few files, unchanged in others
- cleans up some duplicate headers

**Comment by dildick** [comment]
@rekovic Could you sign off, please? Thank you.

**Comment by silviodonato** [comment]
kind reminder @cms-sw/l1-l2

**Comment by rekovic** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**